File: 1889-check-if-number-is-a-sum-of-powers-of-three/check-if-number-is-a-sum-of-powers-of-three.js
/**
 * @param {number} n
 * @return {boolean}
 */
var checkPowersOfThree = function(n) {
    const powers = [];
    let val = 0;
    let i = 0;
    while(val < n) {
        val = Math.pow(3, i++);
        powers.push(val);
    }
    
    while(powers.length) {
        let d = powers.pop();
        if(d <= n) n -= d;
    }



    return n == 0;

};

File: 1889-check-if-number-is-a-sum-of-powers-of-three/check-if-number-is-a-sum-of-powers-of-three.cpp
class Solution {
public:
    bool checkPowersOfThree(int n) {
        


        vector<int> powers;

        for(int i = 0; i <= 14; i++) powers.push_back(pow(3, i));


        for(int i = 14; i >= 0; i--) {
            if(powers[i] <= n) n -= powers[i];
        }

        return n == 0;
        
    }
};